pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionMetaDataManager.java

```diff
@@ -241,20 +245,21 @@ public synchronized boolean containsFunction(String tenant, String namespace, St
      * @param serviceRequest The request
      */
     public void processRequest(MessageId messageId, Request.ServiceRequest serviceRequest) {
-
-        // make sure that processing requests don't happen simultaneously
-        synchronized (this) {
+        try {
             switch (serviceRequest.getServiceRequestType()) {
                 case UPDATE:
-                    this.processUpdate(serviceRequest);
+                    this.processUpdate(serviceRequest.getFunctionMetaData());
                     break;
                 case DELETE:
-                    this.proccessDeregister(serviceRequest);
+                    this.proccessDeregister(serviceRequest.getFunctionMetaData());
                     break;
                 default:
                     log.warn("Received request with unrecognized type: {}", serviceRequest);
             }
+        } catch (IllegalArgumentException e) {
+            // Its ok. Nothing much we can do about it
```

Comment:
Why are we catching and ignoring the exceptions?  Are not the "out of data request errors" thrown as illegal argument exceptions? If we catch them and ignore it, how are to users going to receive the error?

Reply:
processRequest is called by the metadata tailer. Thus it cannot really do much about IllegalArgumentException.
Users will get the IllegalArgumentException from the processUpdate and processDelete from the leader path.